src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -0,0 +1,160 @@
+                )
+                return null; // The user will be warned if they do not have this set and try this scenario with one of the properties set.
+
+            SlnFile sln;
+            try
+            {
+                sln = SlnFileFactory.CreateFromFileOrDirectory(slnPath);
+            }
+            catch (GracefulException)
+            {
+                return null; // This can be called if a solution doesn't exist. MSBuild will catch that for us.
+            }
+
+            List<ProjectInstance> configuredProjects = new List<ProjectInstance>();
+            HashSet<string> configValues = new HashSet<string>();
+            object projectDataLock = new object();
+
+            bool shouldReturnNull = false;
+
+            Parallel.ForEach(sln.Projects.AsEnumerable(), (project, state) =>
+            {
+                const string executableProjectOutputType = "Exe"; // Note that even on Unix when we don't produce exe this is still an exe, same for ASP
+                const string solutionFolderGuid = "{2150E333-8FDC-42A3-9474-1A3956D46DE8}";
+                const string sharedProjectGuid = "{D954291E-2A0B-460D-934E-DC6B0785DB48}";
+
+
+                if (project.TypeGuid == solutionFolderGuid || project.TypeGuid == sharedProjectGuid || !IsValidProjectFilePath(project.FilePath))
+                    return;
+
+                var projectData = TryGetProjectInstance(project.FilePath);
+                if (projectData == null)
+                    return;
+
+                if (projectData.GetPropertyValue("OutputType") == executableProjectOutputType)
+                {
+                    if (ProjectHasUserCustomizedConfiguration(projectData))
+                    {
+                        shouldReturnNull = true;
+                        state.Stop(); // We don't want to override Configuration if ANY project in a sln uses a custom configuration
+                        return;
+                    }
+
+                    string configuration = projectData.GetPropertyValue(slnProjectConfigPropertytoCheck);
+                    if (!string.IsNullOrEmpty(configuration))
+                    {
+                        lock (projectDataLock)
+                        {
+                            configuredProjects.Add(projectData); // we don't care about race conditions here
+                            configValues.Add(configuration);
+                        }
+                    }
+                }
+            });
+
+            if (configuredProjects.Any() && configValues.Count > 1 && !shouldReturnNull)
+            {
+                throw new GracefulException(CommonLocalizableStrings.TopLevelPublishConfigurationMismatchError);
+            }
+
+            return shouldReturnNull || configuredProjects.Count == 0 ? null : configuredProjects.First();
+        }
+
+        /// <summary>
+        /// Provide a CLI input to change configuration based on 
```

Comment:
Nit: I think this is a bit clearer

```suggestion
        /// Returns dotnet CLI command-line parameters (or an empty list) to change configuration based on 
```